lib/cartopy/mpl/gridliner.py

```diff
@@ -219,7 +237,7 @@ def _crs_transform(self):
             transform = transform._as_mpl_transform(self.axes)
         return transform
 
-    def _add_gridline_label(self, value, axis, upper_end):
+    def _add_gridline_label(self, loc, text, axis, edge):
```

Comment:
Loc is no longer in degrees but in projection coordinates. 
The corresponding value in degrees used for the label cannot be exactly inferred by inverse projection because is computed as the intersection an approximate grid line with the edge of the plot.
The formatted label is now directly passed because otherwise it would be neccessary to pass the formatter too.